libraries/audUtils/bidderUtils.js

```diff
@@ -107,23 +108,32 @@ const getBannerDetails = (bidReq) => {
 }
 // Function to get floor price
 const getFloorPrice = (bidReq) => {
-  let bidfloor = bidReq.params && bidReq.params.bid_floor ? bidReq.params.bid_floor : 0
+  let bidfloor = deepAccess(bidReq, 'params.bid_floor', 0);
   return bidfloor;
 }
 // Function to get site object
 const getSiteDetails = (bidderRequest) => {
-  let page = bidderRequest && bidderRequest.refererInfo && bidderRequest.refererInfo.page ? bidderRequest.refererInfo.page : '';
-  let name = bidderRequest && bidderRequest.refererInfo && bidderRequest.refererInfo.domain ? bidderRequest.refererInfo.domain : '';
+  let page = '';
+  let name = '';
+  if (bidderRequest && bidderRequest.refererInfo) {
+    page = bidderRequest.refererInfo.page;
+    name = bidderRequest.refererInfo.domain;
+  }
   return {page: page, name: name};
 }
 // Function to build the user object
 const getUserDetails = (bidReq) => {
   let user = {};
-  if (bidReq.params) {
-    user.id = bidReq.params.user_id ? bidReq.params.user_id : '';
-    user.buyeruid = bidReq.params.buyeruid ? bidReq.params.buyeruid : ''; ;
-    user.keywords = bidReq.params.keywords ? bidReq.params.keywords : '';
-    user.customdata = bidReq.params.customdata ? bidReq.params.customdata : '';
+  if (bidReq && bidReq.ortb2 && bidReq.ortb2.user) {
+    user.id = bidReq.ortb2.user.id ? bidReq.ortb2.user.id : '';
+    user.buyeruid = bidReq.ortb2.user.buyeruid ? bidReq.ortb2.user.buyeruid : '';
+    user.keywords = bidReq.ortb2.user.keywords ? bidReq.ortb2.user.keywords : '';
+    user.customdata = bidReq.ortb2.user.customdata ? bidReq.ortb2.user.customdata : '';
+  } else {
+    user.id = '';
+    user.buyeruid = '';
+    user.keywords = '';
+    user.customdata = '';
```

Comment:
do you need to send an empty object? we'd shave some bytes off requests if this was not included

Reply:
due to some internal checks, we need this properties as blank values if there is no data.